app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -41,20 +40,48 @@ class SearchRepositoriesViewModel(private val repository: GithubRepository) : Vi
     @Volatile
     private var currentQueryValue: String? = null
     @Volatile
-    private var currentSearchResult: Flow<PagingData<Repo>>? = null
+    private var currentSearchResult: Flow<PagingData<UiModel>>? = null
 
     /**
      * Search a repository based on a query string.
      */
-    fun searchRepo(queryString: String): Flow<PagingData<Repo>> {
+    fun searchRepo(queryString: String): Flow<PagingData<UiModel>> {
         val lastResult = currentSearchResult
         if (queryString == currentQueryValue && lastResult != null) {
             return lastResult
         }
         currentQueryValue = queryString
-        val newResult: Flow<PagingData<Repo>> = repository.getSearchResultStream(queryString)
+        val newResult: Flow<PagingData<UiModel>> = repository.getSearchResultStream(queryString)
+                .map { pagingData -> pagingData.map { UiModel.RepoItem(it) } }
+                .map {
+                    it.insertSeparators<UiModel.RepoItem, UiModel> { before, after ->
+                        if (before == null && after is UiModel.RepoItem) {
```

Comment:
`after is UiModel.RepoItem` -> `after != null`

(here and below, you shouldn't need `is` anymore)